core/src/main/java/org/apache/carbondata/core/statusmanager/LoadMetadataDetails.java

```diff
@@ -84,6 +84,9 @@ public void setPartitionCount(String partitionCount) {
   }
 
   public long getLoadEndTime() {
+    if (timestamp == null) {
```

Comment:
Can it be null? why?

Reply:
for stream segment, it is null until the size reach the handoff size

Reply:
Can you replace `-1` with a meaningful constant, to improve readability